src/react/hooks/paragon/useParagonThemeVariants.js

```diff
@@ -0,0 +1,212 @@
+import { useEffect, useState } from 'react';
+
+import { logError, logInfo } from '../../../logging';
+
+import { fallbackThemeUrl, removeExistingLinks } from './utils';
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load each theme variant's CSS, setting the
+ * non-current theme variants as "alternate" stylesheets. That is, the browser will still download
+ * the CSS for the non-current theme variants, but at a lower priority than the current theme
+ * variant's CSS. This ensures that if the theme variant is changed at runtime, the CSS for the new
+ * theme variant will already be loaded.
+ *
+ * @memberof module:React
+ * @param {object} args
+ * @param {object} [args.themeVariants] An object containing the URLs for each supported theme variant, e.g.: `{ light: { url: 'https://path/to/light.css' } }`.
+ * @param {string} [args.currentThemeVariant] The currently applied theme variant, e.g.: `light`.
+ * @param {string} args.onLoad A callback function called when the theme variant(s) CSS is loaded.
+ */
+const useParagonThemeVariants = ({
+  themeVariants,
+  currentThemeVariant,
+  onLoad,
+  onDarkModeSystemPreferenceChange,
+}) => {
+  const [isParagonThemeVariantLoaded, setIsParagonThemeVariantLoaded] = useState(false);
+  const [isBrandThemeVariantLoaded, setIsBrandThemeVariantLoaded] = useState(false);
+
+  useEffect(() => {
+    const changeColorShceme = (colorSchemeQuery) => {
+      onDarkModeSystemPreferenceChange(colorSchemeQuery.matches);
+    };
+    const colorSchemeQuery = window.matchMedia?.('(prefers-color-scheme: dark)');
+    if (colorSchemeQuery) {
+      colorSchemeQuery.addEventListener('change', changeColorShceme);
+    }
+    return () => {
+      if (colorSchemeQuery) {
+        colorSchemeQuery.removeEventListener('change', changeColorShceme);
+      }
+    };
+  }, [onDarkModeSystemPreferenceChange]);
```

Comment:
This name is much better than `someFn`! Looks like a slight typo made it in though.

```suggestion
  useEffect(() => {
    const changeColorScheme = (colorSchemeQuery) => {
      onDarkModeSystemPreferenceChange(colorSchemeQuery.matches);
    };
    const colorSchemeQuery = window.matchMedia?.('(prefers-color-scheme: dark)');
    if (colorSchemeQuery) {
      colorSchemeQuery.addEventListener('change', changeColorScheme);
    }
    return () => {
      if (colorSchemeQuery) {
        colorSchemeQuery.removeEventListener('change', changeColorScheme);
      }
    };
  }, [onDarkModeSystemPreferenceChange]);
```